backend/src/api/courses/courses.api.ts

```diff
@@ -27,6 +27,20 @@ const initCoursesApi: FastifyPluginAsync<Options> = async (fastify, opts) => {
       return rep.status(HttpCode.CREATED).send(course);
     },
   });
+
+  fastify.route({
+    method: HttpMethod.GET,
+    url: CoursesApiPath.ROOT,
```

Comment:
Not really sure, but maybe it would be better to create a route for `courses/search?query=value`
Let's see what coaches think about that

Reply:
Fetching with filtration should go within root route and filtering params in query would provide options - to fetch all data or to filter it; also service method should be `getAll => (opts?: { ...<paginationOpts if provided>, ...<filteringOpts if provided> })`

Reply:
@andanfe So, as I understood, getting all data, getting filtered data and getting paginated data all go in root route and inside the service, we check, if no `paginationOpts` and no `filteringOpts` get all data, if any or both of these data, get related data. Did I understand it correctly?

Reply:
Generally yes, but I think we will check for additional opts on fetching in repository - while building a query
eg
```
const getAllSmth = async (opts?: SomeOptsType) => {
  const { key = '' } = opts ?? {};
  const res = await #SomeModel
    .query()
    .select()
    .where((builder) => { /* it could be defined separately and just passed as cb */
      if (!key) {
        return;
      }
      builder.where({ key });
    })
```

Reply:
Got it, thank you!

Reply:
agreed